contracts/test/TestContracts/InvariantsTestHandler.t.sol

```diff
@@ -2649,33 +2649,35 @@ contract InvariantsTestHandler is Assertions, BaseHandler, BaseMultiCollateralTe
     }
 
     function _aggregateLiquidation(uint256 i, LatestTroveData memory trove, LiquidationTotals storage t) internal {
-        // Coll gas comp
-        uint256 collRemaining = trove.entireColl;
-        uint256 collGasComp = Math.min(collRemaining / COLL_GAS_COMPENSATION_DIVISOR, COLL_GAS_COMPENSATION_CAP);
-        t.collGasComp += collGasComp;
-        collRemaining -= collGasComp;
-
         // Offset debt by SP
         uint256 spRemaining = spBoldDeposits[i] - t.spOffset;
         uint256 spOffset = Math.min(trove.entireDebt, spRemaining > MIN_BOLD_IN_SP ? spRemaining - MIN_BOLD_IN_SP : 0);
         t.spOffset += spOffset;
 
-        // Send coll to SP
+        // Share coll proportionally between SP and redistribution based on offset fraction
+        uint256 collRemaining = trove.entireColl;
         uint256 collSPPortion = collRemaining * spOffset / trove.entireDebt;
-        uint256 spCollGain = Math.min(collSPPortion, spOffset * (_100pct + LIQ_PENALTY_SP[i]) / _price[i]);
+
+        // Deduct coll gas comp from SP portion
+        uint256 collGasComp = Math.min(collSPPortion / COLL_GAS_COMPENSATION_DIVISOR, COLL_GAS_COMPENSATION_CAP);
+        t.collGasComp += collGasComp;
+        collRemaining -= collGasComp;
+
+        // Send remainder of SP portion to SP, capped by liq penalty
+        uint256 spCollGain = Math.min(collSPPortion - collGasComp, spOffset * (_100pct + LIQ_PENALTY_SP[i]) / _price[i]);
```

Comment:
Are the units right here? `spOffset` is in units of BOLD, whereas the first term is in units of collateral? Do we need to scale the second down by the price?

Reply:
It _is_ scaled by the price, isn't it?
```solidity
spOffset * (_100pct + LIQ_PENALTY_SP[i]) / _price[i]
```

Reply:
...yes! Sorry, somehow missed that